### PR TITLE
Deal with the problem that the last block of a SoundFile is not analy…

### PR DESCRIPTION
**Pull request by olafklingt**
…sed.

### CODE REVIEW

**Comment by telephon** [comment]
Why a new pull request? You could just make the changes to the other one (#3). But no problem if you prefer this here.

**Comment by olafklingt** [comment]
Didnt know about that.

**Comment by telephon** [comment]
yes, in the end it's really easy.

**Comment by telephon** [comment]
I have one more little request, but first let me know if you prefer to keep this pull request or push to the other one (for me, both is fine).

**Comment by olafklingt** [comment]
lets use this new one.

File: Classes/SoundFileAnalysis.sc [hunk 1]
```diff
@@ -49,18 +49,33 @@ SoundFileAnalysis {
...
 
 			// get duration and numChannels from soundFile
 			soundFile = SoundFile.openRead(path);
-			analysisDuration = min(duration ?? { soundFile.duration - start }, soundFile.duration);
+
+			options= (serverOptions ?? {ServerOptions()})
+				.verbosity_(-1)
+				.memSize_(8192 * 256) // REALLY NEEDED?
+				.sampleRate_(soundFile.sampleRate);
+			);
```
Comment (telephon) [inline Classes/SoundFileAnalysis.sc]: Here is something I wonder about: if you pass in server options, wouldn't you be surprised that they are overridden? For example, we may want to oversample the analysis and use a higher sample rate than the sound file, etc.

Perhaps it's better to provide a class variable with `defaultServerOptions`? … and then do something like this:
```supercollider
options = serverOptions ? defaultServerOptions;
if(options.sampleRate.isNil) { options = options.copy.sampleRate_(soundFile.sampleRate) };
```

File: HelpSource/Classes/SoundFileAnalysis.schelp [hunk 2]
```diff
@@ -142,6 +144,21 @@ ARGUMENT::maxDataPoints
...
+
+\roundUp analyses the file and adds silence to fill the last block.
+
+nil leaves the duration as given in the argument.
+
+ARGUMENT::serverOptions
+
+link::Classes/ServerOptions:: for the NRT server. A way to deal with the last half block problem is to set the blockSize to 1.
+
+
```
Comment (telephon) [inline HelpSource/Classes/SoundFileAnalysis.schelp:161]: an example would be great here.

File: Classes/SoundFileAnalysis.sc [hunk 3]
```diff
@@ -49,18 +49,33 @@ SoundFileAnalysis {
...
+
+			if(trimDuration.isNil){
+				analysisDuration = duration;
+			}{
+				analysisDuration = min(duration ?? { soundFile.duration - start }, soundFile.duration);
+			};
+			if(trimDuration == \roundUp){
+				analysisDuration=analysisDuration.roundUp(options.blockSize / soundFile.sampleRate);
```
Comment (telephon) [inline Classes/SoundFileAnalysis.sc:79]: spaces before and after `=` would be nice.

**Review by telephon — APPROVED** [review]

**Comment by telephon** [comment]
Thank you!

**Pull request merged**